service/storage_proxy.cc

```diff
@@ -5762,6 +5759,36 @@ inet_address_vector_replica_set storage_proxy::get_live_sorted_endpoints(replica
     return eps;
 }
 
+inet_address_vector_replica_set
+storage_proxy::calculate_target_replicas(
+        db::consistency_level cl,
+        replica::keyspace& ks,
+        inet_address_vector_replica_set live_endpoints,
+        const inet_address_vector_replica_set& preferred_endpoints,
+        db::read_repair_decision repair_decision,
+        gms::inet_address* extra,
+        replica::column_family* cf) const {
+    // Don't access `remote` if this is a local query.
+    if (live_endpoints.empty() || (live_endpoints.size() == 1 && live_endpoints[0] == utils::fb_utilities::get_broadcast_address())) {
```

Comment:
There's a similar check in the next patch. Does it worth having a `bool is_local_query()` helper (template)?